### ISSUE TITLE
Question: InstanceCommand passing data to the execute method

### ISSUE THREAD

**Issue by ewatch** [code16/sharp#237]
Hi all,

the method for the instance command to be excuted looks like this:

```
    /**
     * @param EntityListQueryParams $params
     * @param array $data
     * @return array
     */
    public function execute($instanceId, array $data = []): array
    {
        
    }
```

What is the $data array meant to be used for and how can it be filled ?

**Usecase:**
I have an EntityList filled by an external API.
I want to run an instance command on one of the entities (rows).
Now instead of making again the request to get all the data and then filtering out the data of the entity I'd like to pass data of that particular Entity in the EntityList to the instance command.

I will now implement a workaround to concatenate an artificial instanceId to pass data, which is in my opinion not the best solution.

Do you have any better ideas to implement that or did you have a similiar case ?

Thanks and best regards,
ewatch

**Comment by dvlpp** [comment]
Hi,

`$data` is filled with the form data, when the Command defines one (see https://sharp.code16.fr/docs/guide/commands.html#write-the-command-class).

Your use case seems to be different: if I understood well, you'd like to get the transformed instance (from the `EntityList`) in the `Command`. Sharp will not help you for this: you can maybe, to avoid code duplication, create transformer classes called from the `EntityList` and from the `Command`, or even move the actual transformation to a static function in the `EntityList` and call it from the `Command`. But there is no easy way to avoid the transformation to be computed again: this is a new request.

(Final note: I never seen such a use case, and maybe there is something I misunderstood: I always need the real model in a `Command`, not the transformed instance, and if the `Command` changed the model as it's displayed in the `EntityList`, I simply return a `$this->refresh()` statement (see https://sharp.code16.fr/docs/guide/commands.html#write-the-command-class).

**Comment by ewatch** [comment]
Hi dvlpp,

understood with the `$data` parameter.

As the EntityList is really flexible I used it for a different data source which is a REST API (instead of models from the database).
With this way I made a section in the admin area where a user can select in the EntityList (which is filled by the API) to import data from that API. I wanted to prevent to make another request for the same data that has already been fetched for the EntityList.
